### PR TITLE
Bug 1129306 - Passwords

### PR DESCRIPTION
**Pull request by wesj**
This implements a passwords store using sqlite. It also includes the connection between this and content. That's mostly copied from Firefox's LoginManagerContent.js script. I'll upload a diff to the bug.

My goal for now is to just be careful that 1.) We're never trusting any domains sent from content and 2.) We only fill logins in the top frame for the page. Happy to get lots of feedback on security implications.

Also, this patch lands disabled (the actual saving of form data is commented out in PasswordsManager.swift. Since we don't have any sort of "Do you want to save..." prompt, I decided to just leave this off for now.

### CODE REVIEW

File: Client/Assets/Passwords.js [hunk 1]
```diff
@@ -0,0 +1,699 @@
+/* vim: set ts=2 sts=2 sw=2 et tw=80: */
+/* map Source Code Form is subject to the terms of the Mozilla Public
```
Comment (thebnich) [inline Client/Assets/Passwords.js]: Nit: map -> This

File: Client/Assets/Passwords.js [hunk 2]
```diff
@@ -0,0 +1,699 @@
...
+    return this._getPasswordOrigin(uriString, true);
+  },
+}
+
+function onBlur(event) {
+  LoginManagerContent.onUsernameInput(event)
+}
+
+window.addEventListener("load", function(event) {
+  try {
```
Comment (thebnich) [inline Client/Assets/Passwords.js]: Maybe we could start doing something like <a href="http://pastebin.mozilla.org/8702302">this</a> for all functions defined in content scripts?

Don't have to do this here, just thinking about ways to make this better.

File: Client/Assets/Passwords.js [hunk 3]
```diff
@@ -0,0 +1,699 @@
...
+  if (!gDebug)
+    return;
+
+  var message = generateLogMessage(pieces);
+  // Turn off logging, since its visible to the page
+  // dump(message + "\n");
+  // console.log(message);
+}
+
+var LoginManagerContent = {
```
Comment (thebnich) [inline Client/Assets/Passwords.js]: How much did you have to change from desktop to get this working here? I'd hate to end up with a bunch of different forks of mostly-the-same code that we need to keep in sync. I think @st3fan was working on a way to pull reader code from moz-central -- perhaps we can do the same here.

File: Client/Assets/Passwords.js [hunk 4]
```diff
@@ -0,0 +1,699 @@
...
+  _getRandomId: function() {
+    return Math.round(Math.random() * (Number.MAX_VALUE - Number.MIN_VALUE) + Number.MIN_VALUE).toString()
+  },
+
+  _messages: [ "RemoteLogins:loginsFound",
+               "RemoteLogins:loginsAutoCompleted" ],
```
Comment (thebnich) [inline Client/Assets/Passwords.js]: Unused

**Comment by thebnich** [comment]
Before I look too much at the JS, would you be able to include LoginManagerContent.jsm in an initial commit, then apply these commits on top? Git should then recognize that Passwords.js is a modified version of that file, and it would show a diff from the original. This would be helpful for both reviewing and blame context.

File: Client/Frontend/Browser/PasswordManager.swift [hunk 5]
```diff
@@ -0,0 +1,74 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```
Comment (thebnich) [inline Client/Frontend/Browser/PasswordManager.swift]: Nit: I hope the next version of Xcode fixes these paste issues. Note that several of your new files here don't have these stars lined up.

File: Client/Assets/Passwords.js (hunk 3 above)
Comment (wesj) [inline Client/Assets/Passwords.js]: There's actually a diff of the two files in the bug. Its substantial, but not a crazy amount. I could done a little less if we wanted, but I think Desktop's Password code is actually going to be basically thrown out and rewritten soon.

**Pull request merged**